languages/js/src/Host.ts

```diff
@@ -201,7 +202,14 @@ export class Host {
         return { value: { Boolean: v } };
       case Number.isInteger(v):
         return { value: { Number: { Integer: v } } };
-      case typeof v === 'number' && Number.isFinite(v):
+      case typeof v === 'number':
+        if (v == Infinity) {
+          v = 'Infinity';
+        } else if (v == -Infinity) {
```

Comment:
Inconsequential here, but best practice generally.

```suggestion
        if (v === Infinity) {
          v = 'Infinity';
        } else if (v === -Infinity) {
```